src/lib.rs

```diff
@@ -237,6 +237,152 @@ impl<T: Clone + Integer> Ratio<T> {
     pub fn fract(&self) -> Ratio<T> {
         Ratio::new_raw(self.numer.clone() % self.denom.clone(), self.denom.clone())
     }
+
+    #[cfg(feature = "std")]
+    pub fn as_decimal_string(&self, precision: usize, trail_zeroes: bool) -> std::string::String
+    where
+        T: CheckedMul + fmt::Display
+    {
+        use std::string::String;
+
+        let int_len = {
+            // Build the number 10 from scratch so we can handle both signed and unsigned types (T of i8 or u8)
+            let mut _10 = T::one() + T::one() + T::one(); // 3
+            _10 = _10.clone() * _10 + T::one();           // 10
+
+            let mut ptr: T = T::one();
+            let mut len: usize = 0;
+
+            loop {
+                len += 1;
+
+                ptr = if let Some(p) = ptr.checked_mul(&_10) { p } else { break };
+
+                if ptr > *self.numer() {
+                    break;
+                }
+            }
+
+            len
+        };
+
+        let mut result = String::with_capacity(int_len + precision + if precision > 0 { 1 } else { 0 });
+
+        self.format_as_decimal(&mut result, precision, trail_zeroes).ok();
+
+        result
+    }
+
+    /// Writes the number into the output buffer in decimal representation.  
+    /// No leading zeroes (except a single zero for the integer part)  
+    /// Returns remainder of the division
+    pub fn format_as_decimal(&self, out: &mut fmt::Write, mut precision: usize, trail_zeroes: bool) -> Result<T, fmt::Error>
```

Comment:
Is there much reason to use `fmt::Write` here? It's available in `core`, but AFAICS the only things that implements this trait are `Formatter` and `String` (once you have at least the `alloc` crate). The [`fmt::Write` docs](https://doc.rust-lang.org/core/fmt/trait.Write.html) actively discourage implementing this trait, favoring `io::Write` instead.

Reply:
This was initially designed to work with [fmt::Formatter](https://doc.rust-lang.org/std/fmt/struct.Formatter.html), which implements `fmt::Write` and not io write.
The original idea was to use this function from within the `Display` implementation.
Also, I think this method doesn't have to aim at real `IO`, but rather for string formatting, which IMHO makes `fmt::Write` more appropriate for this particular use case.

Reply:
Though, not a problem to rewrite :)

Reply:
OK, fair enough, especially that `Formatter` doesn't impl `io::Write`.